examples/systemcalls/systemcalls.c

```diff
@@ -1,97 +1,128 @@
- *   and return a boolean true if the system() call completed with success
- *   or false() if it returned a failure
-*/
-
-    return true;
+    if (system(cmd) != -1)
+        return true;
+    else
+        return false;
 }
 
 /**
-* @param count -The numbers of variables passed to the function. The variables are command to execute.
-*   followed by arguments to pass to the command
-*   Since exec() does not perform path expansion, the command to execute needs
-*   to be an absolute path.
-* @param ... - A list of 1 or more arguments after the @param count argument.
-*   The first is always the full path to the command to execute with execv()
-*   The remaining arguments are a list of arguments to pass to the command in execv()
-* @return true if the command @param ... with arguments @param arguments were executed successfully
-*   using the execv() call, false if an error occurred, either in invocation of the
-*   fork, waitpid, or execv() command, or if a non-zero return value was returned
-*   by the command issued in @param arguments with the specified arguments.
-*/
+ * @param count -The numbers of variables passed to the function. The variables are command to execute.
+ *   followed by arguments to pass to the command
+ *   Since exec() does not perform path expansion, the command to execute needs
+ *   to be an absolute path.
+ * @param ... - A list of 1 or more arguments after the @param count argument.
+ *   The first is always the full path to the command to execute with execv()
+ *   The remaining arguments are a list of arguments to pass to the command in execv()
+ * @return true if the command @param ... with arguments @param arguments were executed successfully
+ *   using the execv() call, false if an error occurred, either in invocation of the
+ *   fork, waitpid, or execv() command, or if a non-zero return value was returned
+ *   by the command issued in @param arguments with the specified arguments.
+ */
 
 bool do_exec(int count, ...)
 {
     va_list args;
     va_start(args, count);
-    char * command[count+1];
+    char *command[count + 1];
     int i;
-    for(i=0; i<count; i++)
+    for (i = 0; i < count; i++)
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
     va_end(args);
 
-    return true;
+    pid_t pid;
+    pid = fork();
+
+    if (pid == -1)
+    {
+        return false;
+    }
+    else if (pid == 0)
+    {
+        execv(command[0], command);
+        exit(99);
```

Comment:
@neohere97 suggest a print here since it will be confusing if you ever hit this condition.